services/src/main/java/io/grpc/services/MonitoringService.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2017, Google Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *    * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *    * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ *    * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.grpc.services;
+
+import static com.google.common.base.Preconditions.checkState;
+
+import com.google.instrumentation.stats.RpcConstants;
+import com.google.instrumentation.stats.Stats;
+import com.google.instrumentation.stats.StatsManager;
+import com.google.protobuf.Empty;
+import io.grpc.instrumentation.v1alpha.CanonicalRpcStats;
+import io.grpc.instrumentation.v1alpha.MonitoringGrpc;
+import io.grpc.stub.StreamObserver;
+
+/**
+ * An implementation of the gRPC monitoring service.
+ *
+ * <p>An implementation of {@link StatsManager} must be available at runtime (determined via {@link
+ * Stats#getStatsManager}) or instantiating this service will fail.
+ */
+public final class MonitoringService extends MonitoringGrpc.MonitoringImplBase {
+  private static final StatsManager STATS_MANAGER = Stats.getStatsManager();
```

Comment:
To make it more testable, I would make StatsManager an instance field, pass it to the constructor, call `Stats.getStatsManager()` in `getInstance()` which creates the singleton `instance`, make the constructor package-private, and inject a mock `StatsManager` in the unit test.

Then in `MonitoringServiceTest`, you can verify that `StatsManager.getView()` is called with the expected view descriptors, and that the service is filling out every field as expected (by stubbing `getView()` to return some fake results).